StatsdClient/StatsdExtensions.cs

```diff
@@ -49,6 +49,55 @@ private class StatsBuilderInternal : DynamicObject
       private IStatsd _statsd;
       private List<string> _parts;
       private string _metricType;
+      private static Dictionary<Type, BinaryOperationHandler> _handlerMap = 
+                new Dictionary<Type, BinaryOperationHandler>();
+
+      private delegate void BinaryOperationHandler(IStatsd client, string metricType, string name, object arg);
+
+      static StatsBuilderInternal()
+      {
+        _handlerMap.Add(typeof(int), IntHandler);
+        _handlerMap.Add(typeof(decimal), DoubleHandler);
+        _handlerMap.Add(typeof(double), DoubleHandler);
+        _handlerMap.Add(typeof(float), DoubleHandler);
+      }
+
+      private static void IntHandler(IStatsd client, string metricType, string name, object arg)
+      {
+          var value = Convert.ToInt32(arg);
+          switch (metricType)
+          {
+            case MetricType.COUNT:
+              client.LogCount(name, value);
+              break;
+            case MetricType.GAUGE:
+              client.LogGauge(name, value);
+              break;
+            case MetricType.TIMING:
+              client.LogTiming(name, value);
+              break;
+            case MetricType.SET:
+              client.LogSet(name, value);
+              break;
+          }
+      }
+
+      private static void DoubleHandler(IStatsd client, string metricType, string name, object arg)
```

Comment:
You're potentially losing precision by using a double as the default rather than a decimal.